org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/exporter/AbstractVelocityExporter.java

```diff
@@ -0,0 +1,148 @@
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.Velocity;
+import org.apache.velocity.runtime.RuntimeConstants;
+import org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader;
+import org.apache.velocity.runtime.resource.loader.FileResourceLoader;
+import org.aposin.licensescout.archive.Archive;
+import org.aposin.licensescout.license.License;
+
+/**
+ * The Exporter creates a report listing all archives including license
+ * information.
+ * 
+ * <p>This implementation is a base class for exporters based on Velocity templates
+ * to generate the output.</p>
+ * 
+ * <p>Subclasses have to provide a default template by implementing {@link #getDefaultTemplate()}.</p>
+ * <p>If necessary, subclasses can put additional information to the {@link VelocityContext} by overriding {@link #additionalSetup(VelocityContext, OutputResult)}.</p>
+ */
+public abstract class AbstractVelocityExporter implements IReportExporter {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public final void export(final OutputResult outputResult, final ReportConfiguration reportConfiguration)
+            throws Exception {
+        try (final FileWriter fileWriter = new FileWriter(reportConfiguration.getOutputFile());
+                final BufferedWriter bw = new BufferedWriter(fileWriter);) {
+
+            Velocity.setProperty(RuntimeConstants.RESOURCE_LOADER, "file,classpath");
+            Velocity.setProperty("file.resource.loader.class", FileResourceLoader.class.getName());
+            /*
+             * NOTE: setting the path to empty is necessary because the default is "." (current
+             * directory) and then using absolute path names will not work.
+             */
+            Velocity.setProperty("file.resource.loader.path", "");
+            Velocity.setProperty("classpath.resource.loader.class", ClasspathResourceLoader.class.getName());
+
+            Velocity.init();
+
+            final VelocityContext context = createVelocityContext(outputResult, reportConfiguration);
+            final Template template = getTemplate(reportConfiguration);
+
+            try (final StringWriter sw = new StringWriter();) {
+                if (template != null) {
+                    template.merge(context, sw);
+                }
+                bw.write(sw.getBuffer().toString());
+            }
+        }
+    }
+
+    private VelocityContext createVelocityContext(final OutputResult outputResult,
+                                                  final ReportConfiguration reportConfiguration) {
+        final VelocityContext context = new VelocityContext();
+        final List<Archive> archiveFiles = getSortedArchives(outputResult);
+        context.put("archiveFiles", archiveFiles);
```

Comment:
[Optional - Better readability] move the final list from line 94 to line 84 and remove the empty lines (85 and 93)

Reply:
Kept as it is.